nylon-threads/src/main/java/com/palantir/nylon/threads/VirtualThreads.java

```diff
@@ -0,0 +1,275 @@
+            MethodType threadOfVirtualType = MethodType.methodType(ofVirtual);
+            threadOfVirtual = lookup.findStatic(Thread.class, "ofVirtual", threadOfVirtualType);
+
+            MethodType ofVirtualFactoryType = MethodType.methodType(ThreadFactory.class);
+            ofVirtualFactory = lookup.findVirtual(ofVirtual, "factory", ofVirtualFactoryType);
+
+            MethodType ofVirtualNameStringType = MethodType.methodType(ofVirtual, String.class);
+            ofVirtualNameString = lookup.findVirtual(ofVirtual, "name", ofVirtualNameStringType);
+            MethodType ofVirtualNameStringLongType = MethodType.methodType(ofVirtual, String.class, long.class);
+            ofVirtualNameStringLong = lookup.findVirtual(ofVirtual, "name", ofVirtualNameStringLongType);
+
+            MethodType inheritInheritableThreadLocalsType = MethodType.methodType(ofVirtual, boolean.class);
+            ofVirtualInheritInheritableThreadLocals =
+                    lookup.findVirtual(ofVirtual, "inheritInheritableThreadLocals", inheritInheritableThreadLocalsType);
+
+            MethodType uncaughtExceptionHandlerType = MethodType.methodType(ofVirtual, UncaughtExceptionHandler.class);
+            ofVirtualUncaughtExceptionHandler =
+                    lookup.findVirtual(ofVirtual, "uncaughtExceptionHandler", uncaughtExceptionHandlerType);
+
+            MethodType ofVirtualThreadCreators = MethodType.methodType(Thread.class, Runnable.class);
+            ofVirtualUnstarted = lookup.findVirtual(ofVirtual, "unstarted", ofVirtualThreadCreators);
+            ofVirtualStart = lookup.findVirtual(ofVirtual, "start", ofVirtualThreadCreators);
+
+            MethodType executorsNewThreadPerTaskExecutorType =
+                    MethodType.methodType(ExecutorService.class, ThreadFactory.class);
+            executorsNewThreadPerTaskExecutor = lookup.findStatic(
+                    Executors.class, "newThreadPerTaskExecutor", executorsNewThreadPerTaskExecutorType);
+        }
+
+        @Override
+        public boolean isVirtual(Thread thread) {
+            try {
+                return (boolean) threadIsVirtual.invokeExact(thread);
+            } catch (RuntimeException | Error e) {
+                throw e;
+            } catch (Throwable t) {
+                throw new SafeRuntimeException("failed to invoke 'thread.isVirtual()'", t);
+            }
+        }
+
+        @Override
+        public ExecutorService newThreadPerTaskExecutor(ThreadFactory threadFactory) {
+            try {
+                return (ExecutorService) executorsNewThreadPerTaskExecutor.invoke(threadFactory);
+            } catch (RuntimeException | Error e) {
+                throw e;
+            } catch (Throwable t) {
+                throw new SafeRuntimeException("failed to invoke 'Executors.newThreadPerTaskExecutor'", t);
+            }
+        }
+
+        @Override
+        @SuppressWarnings({"SafeLoggingPropagation", "IllegalSafeLoggingArgument"})
```

Comment:
I should attempt to delete these now, originally I got a strange exception from error-prone